### ISSUE TITLE
Python3 compatibility issues

### ISSUE THREAD

**Issue by jamesobutler** [SimpleITK/SlicerSimpleFilters#17]
There are some compatibilities issues that were found by a user using the recent Slicer nightly which uses Python3. 

https://discourse.slicer.org/t/simple-filters-module-do-not-have-any-filters-in-latest-nightly-version/6787
- On loading module
```log
Error while reading "C:\Program Files\Slicer 4.11.0-2019-05-13\lib\Slicer-4.11\qt-scripted-modules/Resources/json\AbsImageFilter.json". Exception: name 'file' is not defined
Error while reading "C:\Program Files\Slicer 4.11.0-2019-05-13\lib\Slicer-4.11\qt-scripted-modules/Resources/json\AbsoluteValueDifferenceImageFilter.json". Exception: name 'file' is not defined
Error while reading "C:\Program Files\Slicer 4.11.0-2019-05-13\lib\Slicer-4.11\qt-scripted-modules/Resources/json\AcosImageFilter.json". Exception: name 'file' is not defined
.
.
.
```

- Upon selecting one of the filters after changing `file()` usage to `open()`, there were more errors:  
```log
Traceback (most recent call last):
  File "C:/Program Files/Slicer 4.11.0-2019-05-13/bin/../lib/Slicer-4.11/qt-scripted-modules/SimpleFilters.py", line 268, in onFilterSelect
    self.filterParameters.create(json)
  File "C:/Program Files/Slicer 4.11.0-2019-05-13/bin/../lib/Slicer-4.11/qt-scripted-modules/SimpleFilters.py", line 817, in create
    w = self.createIntWidget(member["name"],t)
  File "C:/Program Files/Slicer 4.11.0-2019-05-13/bin/../lib/Slicer-4.11/qt-scripted-modules/SimpleFilters.py", line 963, in createIntWidget
    w.setValue(int(default))
NameError: name 'default' is not defined
```

More issues might be present, but these were the first I have noticed.

**Comment by blowekamp** [comment]
Thank you for reporting the issue. Is this bug in the currently posted nightly build?

**Comment by jamesobutler** [comment]
Yes. Slicer nightly was download from slicer.org today as indicated by
`C:\Program Files\Slicer 4.11.0-2019-05-13` in the errors above.

**Comment by blowekamp** [comment]
It looks like Python3 may not be as forgiving with joining paths on windows with a mix of '\' and '/' slashes. I believe the issue is here on these lines:
https://github.com/SimpleITK/SlicerSimpleFilters/blob/12efaabb0151c270e61bc7a3b3d08ad4c8db8095/SimpleFilters/SimpleFilters.py#L35-L36

and like would only present itself on windows. I would think changing:
```
ICON_DIR = os.path.dirname(os.path.realpath(__file__)) + '/Resources/Icons/'
```
to
```
ICON_DIR = os.path.join( os.path.dirname(os.path.realpath(__file__)), 'Resources', 'Icons')
JSON_DIR = os.path.join( os.path.dirname(os.path.realpath(__file__)), 'Resources, 'json')
```

Are you able to give that change a try?

**Comment by jamesobutler** [comment]
Changing
https://github.com/SimpleITK/SlicerSimpleFilters/blob/12efaabb0151c270e61bc7a3b3d08ad4c8db8095/SimpleFilters/SimpleFilters.py#L105
to use `open()` fixes the initial error.

The second error is something related to these type of calls where "default" is now undefined in python3.  Changing the joining of paths doesn't solve this issue.
https://github.com/SimpleITK/SlicerSimpleFilters/blob/12efaabb0151c270e61bc7a3b3d08ad4c8db8095/SimpleFilters/SimpleFilters.py#L962-L963

**Comment by blowekamp** [comment]
nice job tracking done the file error!

**Comment by jamesobutler** [comment]
Closed by #19.
